System/Win32/NLS.hsc

```diff
@@ -100,26 +348,162 @@ foreign import WINDOWS_CCONV unsafe "windows.h SetLocaleInfoW"
 
 type LCMapFlags = DWORD
 
+-- The version of MSYS2 provided with GHC before GHC 7.10 excludes definitions
+-- of these locale map flag constants
+#if __GLASGOW_HASKELL__ < 710
```

Comment:
move these to a compat file, see main comment

Reply:
I have created compat files. I have to use the CI process to see if they actually work, which may involve some trial and error. I'll post when I think it is working.